metricbeat/module/prometheus/collector/collector.go

```diff
@@ -147,27 +180,36 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 		}
 	}
 
+	return err
+}
+
+// Close stops the metricset
+func (m *MetricSet) Close() error {
+	m.promEventsGen.Stop()
 	return nil
 }
 
-func (m *MetricSet) addUpEvent(eventList map[string]common.MapStr, up int) {
-	metricName := "up"
-	if m.skipFamilyName(metricName) {
-		return
+func (m *MetricSet) upMetricFamily(value float64) *dto.MetricFamily {
+	gauge := dto.Gauge{
+		Value: &value,
 	}
-	upPromEvent := PromEvent{
-		labels: common.MapStr{
-			"instance": m.Host(),
-			"job":      "prometheus",
-		},
+	label1 := dto.LabelPair{
+		Name:  &upMetricInstanceLabel,
+		Value: &m.host,
```

Comment:
Wouldn't be the same if we retrieve the host with `m.Host()`. Wondering if we actually need this extra `host` field for the struct.

Reply:
you cannot use `m.Host()` as a pointer, that's why I create `m.host`. I will add a comment explaining that

Reply:
👍